src/nanoemoji/util.py

```diff
@@ -85,30 +87,120 @@ def file_printer(filename):
             yield partial(print, file=f)
 
 
-def ensure_ttfont_fully_decompiled(ttfont: ttLib.TTFont):
-    # A TTFont might be opened lazily and some tables only partially decompiled.
-    # So for this to work on any TTFont, we first compile everything to a temporary
-    # stream then reload with lazy=False. Input font is modified in-place.
+def _assert_all_keys_loaded(font: ttLib.TTFont):
+    not_loaded = sorted(t for t in font.keys() if not font.isLoaded(t))
+    assert (
+        not not_loaded
+    ), f"Everything should be loaded, following aren't: {not_loaded}"
+
+
+def _reload(font: ttLib.TTFont, lazy: bool = True):
+    # Stream font to memory and load it back again
     tmp = BytesIO()
-    ttfont.save(tmp)
+    font.save(tmp)
     tmp.seek(0)
-    ttfont2 = ttLib.TTFont(tmp, lazy=False)
-    for tag in ttfont2.keys():
-        table = ttfont2[tag]
+    font2 = ttLib.TTFont(tmp, lazy=lazy)
+
+
+def load_fully(font: Union[Path, ttLib.TTFont]) -> ttLib.TTFont:
+    if isinstance(font, Path):
+        font = ttLib.TTFont(str(font), lazy=False)
+    else:
+        # A TTFont might be opened lazily and some tables only partially decompiled.
+        # If so, reload it
+        if font.lazy:
+            font = _reload(font, lazy=False)
+
+    # one would think lazy=False meant tags would report isLoaded true
+    # one would be wrong, following fails:
+    # _assert_all_keys_loaded(font)
+
+    # OK, for those of you that are STILL lazy just bloody load
+    for tag in font.keys():
+        table = font[tag]
         # cmap is exceptional in that it always loads subtables lazily upon getting
         # their attributes, no matter the value of TTFont.lazy option.
         # TODO: remove this hack once fixed in fonttools upstream
         if tag == "cmap":
             _ = [st.cmap for st in table.tables]
-        ttfont[tag] = table
 
+    _assert_all_keys_loaded(font)
+
+    # post has a cool pattern of raising errors if it has no glyphOrder attribute
+    # font isn't all that useful w/o glyph order so lets confirm we're ok on that front
+    # TODO contemplate whether this is simply a FontTools bug
+    font.getGlyphOrder()
```

Comment:
Why this?

Reply:
The comment aimed to explain but I guess it doesn't. It's possible to create a font where getGlyphOrder will fail after a reorder attempt and it's really annoying when that happens later in some random spot. I prefer to confirm it won't kerplode right here.

Reply:
can you tell me exactly how to reproduce this? It's certainly a bug that we should fix upstream

Reply:
this comment

> post has a cool pattern of raising errors if it has no glyphOrder attribute

is misleading. One is not supposed to ever access the post.glyphOrder directly, only via TTFont.getGlyphOrder()

Reply:
Ah that's my mistake in phrasing. So what ends up happening is you call TTFont::getGlyphOrder and it crashes _in_ post. It's not caused by a direct call into post.

Reply:
then that's even more of a bug!

Reply:
Now if only I'd been smart enough to record how to achieve that state...

Reply:
I think maybe this doesn't matter as I've migrated to the new toys you added to fontTools. If I find a new repro I'll file a bug.